kyuubi-common/src/main/scala/org/apache/kyuubi/engine/ComponentVersion.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.engine
+
+/**
+ * Encapsulate a component (Kyuubi/Spark/Hive/Flink) version for the convenience of version checks.
+ */
+class ComponentVersion(versionString: String) {
+
+  val (majorVersion, minorVersion, patchVersion) =
+    """^(\d+)\.(\d+)\.(.*?)$""".r.findFirstMatchIn(versionString) match {
+      case Some(m) =>
+        (m.group(1).toInt, m.group(2).toInt, m.group(3))
+      case None =>
+        throw new IllegalArgumentException(s"Tried to parse '$versionString' as a project" +
+          s" version string, but it could not find the major, minor and patch version numbers.")
+    }
+}
+
+object ComponentVersion {
+
+  def isVersionAtMost(targetVersionString: String, runtimeVersionString: String): Boolean = {
```

Comment:
define `def isVersionAtMost(targetVersionString: String)` in the companion class

Reply:
Agreed. 

The following API is more intuitive.
```
ComponentVersion("3.2.2").isVersionAtMost("3.3")
```